internals/overlord/overlord.go

```diff
@@ -379,15 +371,34 @@ func (o *Overlord) Loop() {
 			}
 		}
 	})
+	o.ensureWaitRun()
 }
 
 func (o *Overlord) ensureDidRun() {
-	atomic.StoreInt32(&o.ensureRun, 1)
+	select {
+	case <-o.ensureRun:
+		// Already closed. Ensure already ran at least once.
+	default:
+		close(o.ensureRun)
+	}
 }
 
-func (o *Overlord) CanStandby() bool {
-	run := atomic.LoadInt32(&o.ensureRun)
-	return run != 0
+// ensureWaitRun waits until StateEngine.Ensure() was called at least once.
+func (o *Overlord) ensureWaitRun() {
```

Comment:
This function waits, so it would make more sense if it were called `waitEnsureRun`, that is, "wait till ensure has run". The name `ensureWaitRun` sounds more like "ensure that the wait has run".

Reply:
Improved as per proposal, thank you. Resolving.